ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -104,28 +104,43 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    this.deltas.addAll(filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)));
+    int bucketId = AcidUtils.parseBucketId(path);
+    AcidUtils.ParsedDeltaLight parentDelta = AcidUtils.ParsedDeltaLight.parse(getPath().getParent());
```

Comment:
could be refactored using static import:
````
ParsedDeltaLight pd = ParsedDeltaLight.parse(path.getParent())
````

Reply:
Hi @deniskuzZ,
Many tests failed with the change of using AcidUtils.ParsedDeltaLight.parse() instead of AcidUtils.parseBaseOrDeltaBucketFilename(). As I understand the split is not always a delta folder, it can be some older format not supported by ParsedDeltaLight. I saw that ParsedDeltaLight.parse() is used in some cases internally in AcidUtils.parseBaseOrDeltaBucketFilename(), but not always. Can you please advise if I should revert to using AcidUtils.parseBaseOrDeltaBucketFilename() that worked in all cases or there is some better way?

Reply:
reverted to previous version without ParsedDeltaLight.parse(path.getParent())